source/dynamical_systems/include/dynamical_systems/IDynamicalSystem.hpp

```diff
@@ -116,8 +118,16 @@ class IDynamicalSystem {
    */
   void assert_parameter_valid(const std::string& name, state_representation::StateType state_type);
 
+  /**
+   * @brief Validate and set a parameter of the dynamical system. Internal
+   * function, to be redefined based on the dynamical system, called
+   * by the set_parameter function.
+   * @param parameter The parameter to be validated
+   */
+  virtual void validate_parameter(const std::shared_ptr<state_representation::ParameterInterface>& parameter) = 0;
```

Comment:
Initially I thought `set_parameter` would call `validate_parameter` and then subsequently do some generic / templated setting, but the behaviour of the virtual method `validate_parameter` doing the actual setting on a class-by-class basis makes a lot more sense.

However, I would recommend we rename this to `validate_and_set_parameter` to make it explicitly clear that the "setting" has to be done in the derived class!

Reply:
totally agree